src/System.Security.SecureString/src/System/Security/SafeBSTRHandle.cs

```diff
@@ -11,16 +11,24 @@ internal sealed class SafeBSTRHandle : SafeBuffer
     {
         internal SafeBSTRHandle() : base(true) { }
 
-        internal static SafeBSTRHandle Allocate(string src, uint len)
+        internal static SafeBSTRHandle Allocate(string src, uint lenInChars)
         {
-            SafeBSTRHandle bstr = Interop.OleAut32.SysAllocStringLen(src, len);
-            bstr.Initialize(len * sizeof(char));
+            SafeBSTRHandle bstr = Interop.OleAut32.SysAllocStringLen(src, lenInChars);
+            if (bstr.IsInvalid) // SysAllocStringLen returns a NULL ptr when there's insufficient memory
+            {
+                throw new OutOfMemoryException();
+            }
+            bstr.Initialize(lenInChars * sizeof(char));
             return bstr;
         }
 
         internal static SafeBSTRHandle Allocate(IntPtr src, uint lenInBytes)
         {
             SafeBSTRHandle bstr = Interop.OleAut32.SysAllocStringLen(src, lenInBytes / sizeof(char));
```

Comment:
I'll add an assert to verify that it always represents a number of characters, i.e. that lenInBytes % sizeof(char) == 0.